configstack/running_module.go

```diff
@@ -184,13 +202,23 @@ func (module *runningModule) runNow() error {
 	}
 }
 
+var separator = strings.Repeat("-", 132)
+
 // Record that a module has finished executing and notify all of this module's dependencies
 func (module *runningModule) moduleFinished(moduleErr error) {
-	if moduleErr == nil {
-		module.Module.TerragruntOptions.Logger.Printf("Module %s has finished successfully!", module.Module.Path)
-	} else {
-		module.Module.TerragruntOptions.Logger.Printf("Module %s has finished with an error: %v", module.Module.Path, moduleErr)
+	status := "successfully!"
+	if moduleErr != nil {
+		status = fmt.Sprintf("with an error: %v", moduleErr)
+	}
+
+	module.Mutex.Lock()
+	defer module.Mutex.Unlock()
```

Comment:
Are you saying `Printf` calls are being interleaved on a per-character basis? I've never seen that and we do a TON of logging in parallel in Go as part of our automated tests in dozens of projects... Is that a Windows-specific issue? Or a result of calling `fmt.Fprintf` as opposed to `Logger.Printf`?